### PR TITLE
[HttpKernel] Turn bad hosts into 400 instead of 500

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Comment by sroze** [comment]
Definitely makes sense 👍

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

File: src/Symfony/Component/HttpKernel/Tests/EventListener/RouterListenerTest.php [hunk 1]
```diff
@@ -155,4 +155,19 @@ public function getLoggingParameterData()
             array(array(), 'Matched route "n/a".'),
         );
     }
+
+    /**
+     * @expectedException \Symfony\Component\HttpKernel\Exception\BadRequestHttpException
+     */
+    public function testSubRequestWithBadHost()
```
Comment (lyrixx) [inline src/Symfony/Component/HttpKernel/Tests/EventListener/RouterListenerTest.php]: It's not about sub request. But master request

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpKernel/Tests/EventListener/RouterListenerTest.php]: "Sub" removed, thanks

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpKernel/Tests/EventListener/RouterListenerTest.php]: oups, it's already merged :) PR welcome then :P

**Pull request merged**